src/processor/model/otel_transform.rs

```diff
@@ -0,0 +1,127 @@
+use crate::processor::model::Value::{BoolValue, BytesValue, DoubleValue, IntValue, StringValue};
+use crate::processor::model::{ResourceSpans, Span};
+use std::sync::{Arc, Mutex};
+
+pub fn transform(rs: opentelemetry_proto::tonic::trace::v1::ResourceSpans) -> ResourceSpans {
+    let mut resource_span = ResourceSpans {
+        resource: Arc::new(Mutex::new(None)),
+        scope_spans: Arc::new(Mutex::new(vec![])),
+        _schema_url: Arc::new(Mutex::new("".to_string())),
+    };
+    if rs.resource.is_some() {
+        let resource = rs.resource.unwrap();
+        let kvs = build_rotel_sdk_resource(resource);
+        let res = Arc::new(Mutex::new(Some(crate::processor::model::Resource {
+            attributes: Arc::new(Mutex::new(kvs.clone())),
+            dropped_attributes_count: Arc::new(Mutex::new(0)),
+        })));
+        resource_span.resource = res.clone()
+    }
+    let mut scope_spans = vec![];
+    for ss in rs.scope_spans {
+        let scope_span = crate::processor::model::ScopeSpans {
+            spans: Arc::new(Mutex::new(vec![])),
+            schema_url: Arc::new(Mutex::new("".to_string())),
+        };
+        for s in ss.spans {
+            let span = Span {
+                trace_id: s.trace_id,
+                span_id: s.span_id,
+                trace_state: s.trace_state,
+                parent_span_id: s.parent_span_id,
+                flags: s.flags,
+                name: s.name,
+                kind: s.kind,
+                start_time_unix_nano: s.start_time_unix_nano,
+                end_time_unix_nano: s.end_time_unix_nano,
+                attributes: Arc::new(Mutex::new(vec![])),
```

Comment:
Can this use the attributes conversion code for a resource or would that need to be separate?

Reply:
Same as above, added TODO